src/util/audio/AudioQueue.h

```diff
@@ -100,7 +100,7 @@ class AudioQueue : protected std::deque<T>
 
 		{
 			std::lock_guard<std::mutex> lock(internalLock);
-			returnBufferLength = std::min(nSamples, std::deque<T>::size() - std::deque<T>::size() % this->channels);
+			returnBufferLength = std::min(nSamples, (unsigned long)(std::deque<T>::size() - std::deque<T>::size() % this->channels));
```

Comment:
`returnBufferLength = std::min<unsigned>(nSamples, std::deque<T>::size() - std::deque<T>::size() % this->channels));` should work also. 
But I would rather change returnBufferLength to size_t.